__tests__/app.test.js

```diff
@@ -0,0 +1,351 @@
+        return request(app)
+            .patch("/api/articles/1")
+            .send(testPatchMinus)
+            .expect(200)
+            .then(({ body }) => {
+                const updatedVotes = body.article.votes
+                const expectedVotes = articleId1.votes + testPatchMinus.inc_votes
+                expect(updatedVotes).toBe(expectedVotes)
+        })
+    })
+    test("400 - responds with error message when patch is not a number", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send(errorPatch)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+        })
+    })
+    test("400 - responds with error message when article id is invalid", () => {
+        return request(app)
+            .patch("/api/articles/one")
+            .send(testPatch)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+        })
+    })
+    test("400 - responds with error message when patch is empty", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+            })
+    })
+    test("404 - responds with error message when article id does not exist", () => {
+        return request(app)
+            .patch("/api/articles/99999")
+            .send(testPatch)
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("no article found with ID 99999")
+            })
+    })
+})
+
+describe('DELETE /api/comments/:comment_id', () => {
+    const commentId = 1
+    test("204 - responds with empty string after deleting comment", () => {
+        return request(app)
+            .delete(`/api/comments/${commentId}`)
+            .expect(204)
+            .then(({ text }) => {
+                expect(text).toBe("")
+        })
+    })
+    test("400 - responds with error message when comment id is invalid", () => {
+        return request(app)
+            .delete("/api/comments/one")
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+            })
+    })
+    test("404 - responds with error message when comment id does not exist", () => {
+        return request(app)
+            .delete("/api/comments/99999")
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("no comment found with ID 99999")
+            })
+    })
+})
+
+describe('GET /api/users', () => {
+    test("200 - respond with all users", () => {
+        return request(app)
+        .get("/api/users")
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.users).toHaveLength(4)
+            expect(typeof body.users).toEqual("object")
```

Comment:
This line is unnecessary and somewhat confusing as it is an array (which happens to be an object). We don't need this expect because the forEach will check it for us